osu.Server.Queues.ScoreStatisticsProcessor/LegacyRulesetHelper.cs

```diff
@@ -32,5 +37,26 @@ public static Ruleset GetRulesetFromLegacyId(int legacyId)
                     throw new ArgumentException($"Invalid ruleset ID: {legacyId}", nameof(legacyId));
             }
         }
+
+        public static DifficultyAttributes CreateDifficultyAttributes(int legacyId)
+        {
+            switch (legacyId)
+            {
+                case 0:
+                    return new OsuDifficultyAttributes();
+
+                case 1:
+                    return new TaikoDifficultyAttributes();
+
+                case 2:
+                    return new CatchDifficultyAttributes();
+
+                case 3:
+                    return new ManiaDifficultyAttributes();
+
+                default:
+                    throw new ArgumentException($"Invalid ruleset ID: {legacyId}", nameof(legacyId));
+            }
+        }
```

Comment:
Probably should be a method on `Ruleset` at some point, but this is probably fine for now.